misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/IterableComparableAssertionsSpec.kt

```diff
@@ -0,0 +1,98 @@
+import ch.tutteli.atrium.specs.AssertionCreatorSpec
+import ch.tutteli.atrium.specs.Feature0
+import ch.tutteli.atrium.specs.Fun1
+import ch.tutteli.atrium.specs.SubjectLessSpec
+import ch.tutteli.atrium.specs.adjustName
+import ch.tutteli.atrium.specs.describeFunTemplate
+import ch.tutteli.atrium.specs.forAssertionCreatorSpec
+import ch.tutteli.atrium.specs.forSubjectLess
+import ch.tutteli.atrium.specs.lambda
+import ch.tutteli.atrium.specs.name
+import ch.tutteli.atrium.specs.toBeDescr
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.Suite
+
+abstract class IterableComparableAssertionsSpec(
+    maxFeature: Feature0<Iterable<Int>, Int>,
+    max: Fun1<Iterable<Int>, Expect<Int>.() -> Unit>,
+    nullableMaxFeature: Feature0<Iterable<Int?>, Int>,
+    nullableMax: Fun1<Iterable<Int?>, Expect<Int>.() -> Unit>,
+    describePrefix: String = "[Atrium] "
+) : Spek({
+
+    include(object : SubjectLessSpec<Iterable<Int>>(describePrefix,
+        maxFeature.forSubjectLess().adjustName { "$it feature" },
+        max.forSubjectLess { toBe(1) }
+    ) {})
+
+    include(object : AssertionCreatorSpec<Iterable<Int>>(
+        describePrefix, listOf(1),
+        max.forAssertionCreatorSpec("$toBeDescr: 1") { toBe(1) }
+    ) {})
+
+    fun describeFun(vararg funName: String, body: Suite.() -> Unit) =
+        describeFunTemplate(describePrefix, funName, body = body)
+
+    describeFun("val ${maxFeature.name}") {
+        val maxVal = max.lambda
+        val nullableMaxVal = nullableMax.lambda
+        checkMax( { assertion -> maxVal(assertion) }, { assertion -> nullableMaxVal(assertion) })
+    }
+
+    describeFun("fun ${max.name}") {
+        val maxFun = maxFeature.lambda
+        val nullableMaxFun = nullableMaxFeature.lambda
+        checkMax( { assert -> maxFun().assert() }, { assert -> nullableMaxFun().assert() })
+    }
+})
+
+private fun Suite.checkMax(
+    testMax: Expect<Iterable<Int>>.(Expect<Int>.() -> Unit) -> Unit,
+    testNullableMax: Expect<Iterable<Int?>>.(Expect<Int>.() -> Unit) -> Unit
+) {
+    val iterableOfNulls = expect(listOf<Int?>(null) as Iterable<Int?>)
+
+    val emptyIterable = expect(emptyList<Int>() as Iterable<Int>)
+
+    val filledIterable = expect(listOf(1, 2) as Iterable<Int>)
+
+    context("list with 1 and 2") {
+        it("toBe(2) holds") {
+            filledIterable.testMax { toBe(2) }
+        }
+        it("toBe(1) fails") {
+            expect {
+                filledIterable.testMax { toBe(1) }
+            }.toThrow<AssertionError> {
+                messageContains("max: 2")
+            }
+        }
+    }
+
+    context("empty list") {
+        it("fails warning about empty iterable") {
+            expect {
+                emptyIterable.testMax { toBe(1) }
+            }.toThrow<AssertionError> {
+                messageContains("cannot be determined, empty Iterable")
+            }
+        }
+    }
+
+    context("list of nulls") {
```

Comment:
Null is a good use case but actually not possible as Kotlin's max only accepts `T : Comparable<T>` and comparable is invariant (as far as I know, check if you think I am wrong)